### PR TITLE
MRG: update to plot_2d fixes and tests

### PR DESCRIPTION
**Pull request by matthew-brett**
Some more refactoring of plot_2d docstrings and another sampling_shape bugfix.
Refactor Ariel's smoke tests to work on travis.

### CODE REVIEW

**Comment by matthew-brett** [comment]
Omar - if the tests pass, and these changes look right to you, feel free to go ahead and merge.

File: dipy/viz/regtools.py [hunk 1]
```diff
@@ -90,162 +93,171 @@ def draw_lattice_2d(nrows, ncols, delta):
...
 
-    #Draw the squares on the output grid
-    X1,X0 = np.mgrid[0:inverse_grid_shape[0], 0:inverse_grid_shape[1]]
-    lattice_out=draw_lattice_2d((inverse_grid_shape[0] + delta) / (delta + 1), 
-                                (inverse_grid_shape[1] + delta) / (delta + 1),
-                                delta)
-    lattice_out=lattice_out[0:inverse_grid_shape[0], 0:inverse_grid_shape[1]]
+    # Draw the squares on the output grid
+    lattice_out = draw_lattice_2d(
+        (inverse_grid_shape[0] + delta) // (delta + 1),
```
Comment (matthew-brett) [inline dipy/viz/regtools.py]: Omar - is this correct?  Integer division (rounding down)?

File: dipy/viz/regtools.py [hunk 2]
```diff
@@ -90,162 +93,171 @@ def draw_lattice_2d(nrows, ncols, delta):
...
+        world_to_image = np.linalg.inv(direct_grid_affine)
 
-    #Draw the squares on the input grid
-    X1,X0 = np.mgrid[0:direct_grid_shape[0], 0:direct_grid_shape[1]]
-    lattice_in=draw_lattice_2d((direct_grid_shape[0] + delta) / (delta + 1), 
-                               (direct_grid_shape[1] + delta) / (delta + 1),
-                               delta)
-    lattice_in=lattice_in[0:direct_grid_shape[0], 0:direct_grid_shape[1]]
+    # Draw the squares on the input grid
+    lattice_in = draw_lattice_2d((direct_grid_shape[0] + delta) // (delta + 1),
```
Comment (matthew-brett) [inline dipy/viz/regtools.py]: Ditto - is integer division the right thing to do here?

File: dipy/viz/regtools.py (hunk 1 above)
Comment (omarocegueda) [inline dipy/viz/regtools.py]: sorry matthew, i dont have my computer right now, i will come back within one hour an take a look!

-----Original Message-----

**Comment by matthew-brett** [comment]
Tests passing, this one is ready to go from my point of view.

File: dipy/viz/regtools.py (hunk 1 above)
Comment (omarocegueda) [inline dipy/viz/regtools.py]: Hi @matthew-brett, =)
thank you very much for reviewing this!, this is correct, it is actually integer division rounding up: (x+(n-1))//n is division by n rounding up. This considers the case delta=0 (the degenerate case where there is no separation between division lines).

File: dipy/viz/regtools.py (hunk 2 above)
Comment (omarocegueda) [inline dipy/viz/regtools.py]: Yes, it is safer not to deal with floating point operations when possible (we might get different results depending on the precision of the floats being used, like the double vs. extended discussion we had some time ago)

**Pull request merged**